sonic_db_config/db_config.go

```diff
@@ -194,6 +195,166 @@ func GetDbTcpAddr(db_name string, ns string) (addr string, err error) {
 	return hostname + ":" + strconv.Itoa(port), err
 }
 
+func CheckDbMultiInstance() (ret bool, err error) {
+	if !sonic_db_init {
+		err = DbInit()
+		if err != nil {
+			return false, err
+		}
+	}
+	defer CatchException(&err)
+	dbkey_vec := swsscommon.SonicDBConfigGetDbKeys()
+	length := int(dbkey_vec.Size())
+	// If there are more than one instances, this means that SONiC is using multi-namespace or multi-container
+	return length > 1, err
+}
+
+func GetDbNonDefaultInstances() (dbkey_list []swsscommon.SonicDBKey, err error) {
+	if !sonic_db_init {
+		err = DbInit()
+		if err != nil {
+			return dbkey_list, err
+		}
+	}
+	defer CatchException(&err)
+	dbkey_vec := swsscommon.SonicDBConfigGetDbKeys()
+	// Translate from vector to array
+	length := int(dbkey_vec.Size())
+	for i := 0; i < length; i += 1 {
+		dbkey := dbkey_vec.Get(i)
+		if dbkey.GetNetns() == SONIC_DEFAULT_NAMESPACE && dbkey.GetContainerName() == SONIC_DEFAULT_CONTAINER {
+			continue
+		}
+		dbkey_list = append(dbkey_list, dbkey)
+	}
+	return dbkey_list, err
+}
+
+func GetDbAllInstances() (dbkey_list []swsscommon.SonicDBKey, err error) {
+	if !sonic_db_init {
+		err = DbInit()
+		if err != nil {
+			return dbkey_list, err
+		}
+	}
+	defer CatchException(&err)
+	dbkey_vec := swsscommon.SonicDBConfigGetDbKeys()
+	// Translate from vector to array
+	length := int(dbkey_vec.Size())
+	for i := 0; i < length; i += 1 {
+		dbkey := dbkey_vec.Get(i)
+		dbkey_list = append(dbkey_list, dbkey)
+	}
+	return dbkey_list, err
+}
+
+func GetDbInstanceFromTarget(ns string, container string) (dbkey swsscommon.SonicDBKey, ret bool) {
+	dbkey = swsscommon.NewSonicDBKey()
+	dbkey.SetNetns(ns)
+	dbkey.SetContainerName(container)
+	_, err := GetDbListByDBKey(dbkey)
+	if err != nil {
+		return nil, false
+	}
+	return dbkey, true
+}
+
+func GetDbListByDBKey(dbkey swsscommon.SonicDBKey) (db_list []string, err error) {
+	if !sonic_db_init {
+		err = DbInit()
+		if err != nil {
+			return db_list, err
+		}
+	}
+	defer CatchException(&err)
+	db_vec := swsscommon.SonicDBConfigGetDbList(dbkey)
+	// Translate from vector to array
+	length := int(db_vec.Size())
+	for i := 0; i < length; i += 1 {
+		dbname := db_vec.Get(i)
+		db_list = append(db_list, dbname)
+	}
+	return db_list, err
+}
+
+func GetDbSeparatorByDBKey(db_name string, dbkey swsscommon.SonicDBKey) (separator string, err error) {
```

Comment:
>GetDbSeparatorByDBKey [](http://example.com/codeflow?start=5&length=21)

Many functions below are 1:1 mapping of swsscommon functions. Why not directly using swsscommon functions? What is the value of the extra wrapper?

Reply:
This wrapper can translate swsscommon exception to error and invoke swsscommon initialize API.